content/src/content/docs/learn/tutorials/hooking-up-a-database.mdx

```diff
@@ -0,0 +1,195 @@
+---
+type: Tutorial
+title: Hooking up a database
+tags:
+  - some
+  - tags
+sidebar:
+  order: 3
+---
+
+The next endpoint we're going to migrate is one that lists all of the users in a
+database. For illustrative purposes, we're going to use SQLite using Bun's
+built-in connector, but the technique we'll cover will work for
+whatever database you're using, and however you're connecting to it.
+
+First, let's take a look at our app in full with our new endpoint added to
+it:
+
+```ts twoslash {8-9,13-17} title=index.ts
+import express from "express"
+
+import { HttpRouter, HttpServer, HttpServerResponse } from "@effect/platform"
+import { NodeHttpServer, NodeRuntime } from "@effect/platform-node"
+import { Effect, Layer } from "effect"
+import { createServer } from "node:http"
+
+import { Database } from "bun:sqlite";
+const db = new Database("sqlite.db");
+
+const app = express();
+
+app.get("/users", (req, res) => {
+  res.send({
+    users: db.prepare("SELECT * FROM users").all(),
+  });
+});
+
+export const health = Effect.sync(() => {
+  return HttpServerResponse.text("ok")
+})
+
+const router = HttpRouter.empty.pipe(
+  HttpRouter.get("/health", health),
+)
+
+NodeRuntime.runMain(
+  Layer.launch(
+    Layer.provide(
+      HttpServer.serve(router),
+      NodeHttpServer.layer(() => createServer(app), { port: 3000 }),
+    ),
+  ),
+)
```

Comment:
Since you've already taught folks about `.pipe`, you could re-write this to make it flow sequentially:

```ts
HttpServer.server(router).pipe(
  Layer.provide(NodeHttpServer.layer(() => createServer(app), { port: 3000 })),
  Layer.launch,
  NodeRuntime.runMain
)
```